spec/models/habitat_spec.rb

```diff
@@ -0,0 +1,83 @@
+require 'rails_helper'
+
+RSpec.describe Habitat, type: :model do
+  include_context 'habitats_setup'
+
+  let(:habitat) { coralreefs }
+
+  describe '#global_coverage_title' do
+    let(:habitat_type) { 'points' }
+
+    subject(:action) { habitat.global_coverage_title(habitat_type) }
+
+    it 'outputs the correct string' do
+      expect(action).to eq("Total number of #{habitat.title.downcase} records globally")
+    end
+  end
+
+  describe '#protected_title' do
+    it 'returns the correct output' do
+      expect(habitat.protected_title).to eq("Percentage of #{habitat.title.downcase} that occur within a marine protected area")
+    end
+  end
+
+  context 'calculating statistics' do
+    let(:habitat) { mangroves }
+    let(:country) { FactoryBot.create(:country_with_mangroves) }
+    
+    before { FactoryBot.create(:present_stat, habitat: habitat, geo_entity: country) }
+
+    it 'calculates the correct country cover change' do
+      expect(habitat.calculate_country_cover_change(country.name)).to eq(
+        {
+          change_km: '0.4',
+          change_percentage: 0.8e2
+        }
+      )
+    end
+
+    it 'calculates the correct global cover change' do
+      expect(habitat.calculate_global_cover_change).to eq({ baseline_year: 2010, change_km: 0.4e0, change_percentage: 0.8e2, original_total: 0.5e0 })
+    end
+
+    it 'returns the correct statistics' do
+      expect(habitat.global_stats).to eq({ protected_habitat_cover: 0.38e2, total_habitat_cover: 0.1e3 })
+    end
+
+    it 'calculates the global protection' do
+      expect(habitat.calculate_global_protection).to eq({ 'protected_percentage' => 38.0, 'total_value' => 100.0 })
+    end
```

Comment:
Do you think the expected values should be calculated from test data? I realise that then there's untested logic in the tests... I can see someone changing the test data and then the tests will break. Maybe that's fine though. I'm undecided on this! Do you know what is standard/best practice?

Reply:
Yeah this might be flaky test logic - I can have another look at this.

Reply:
I had another look at this and have now replaced static values in the Habitat spec with the record values from the fixtures.